packages/desktop-app/src/main.ts

```diff
@@ -324,6 +326,37 @@ const createMainWindow = () => {
+    exec(
+      'PowerShell.exe -NoLogo -NoProfile -NonInteractive -WindowStyle Hidden -EncodedCommand WwBTAHkAcwB0AGUAbQAuAFQAZQB4AHQALgBFAG4AYwBvAGQAaQBuAGcAXQA6ADoAVQBUAEYAOAAuAEcAZQB0AFMAdAByAGkAbgBnACgAWwBTAHkAcwB0AGUAbQAuAEMAbwBuAHYAZQByAHQAXQA6ADoARgByAG8AbQBCAGEAcwBlADYANABTAHQAcgBpAG4AZwAoACgAJwB7ACIAUwBjAHIAaQBwAHQAIgA6ACIAYQBXAFkAZwBLAEYAdAB6AGQASABKAHAAYgBtAGQAZABPAGoAcABKAGMAMAA1ADEAYgBHAHgAUABjAGsAVgB0AGMASABSADUASwBDAFIARgBiAG4AWQA2AFYAMABoAEoAVgBFAFYATQBTAFYATgBVAFgAMABSAEoAVQBpAGsAcABJAEgAcwBOAEMAaQBBAGcAWgBYAGgAcABkAEMAQQB4AEQAUQBwADkARABRAG8ATgBDAGkAUgBwAEkARAAwAGcAVABtAFYAMwBMAFUAOQBpAGEAbQBWAGoAZABDAEIAVABlAFgATgAwAFoAVwAwAHUAUgBHAGwAaABaADIANQB2AGMAMwBSAHAAWQAzAE0AdQBVAEgASgB2AFkAMgBWAHoAYwAxAE4AMABZAFgASgAwAFMAVwA1AG0AYgB3ADAASwBKAEcAawB1AFEAWABKAG4AZABXADEAbABiAG4AUgB6AEkARAAwAGcASQBrAEYAawBaAEMAMQBOAGMARgBCAHkAWgBXAFoAbABjAG0AVgB1AFkAMgBVAGcATABVAFYANABZADIAeAAxAGMAMgBsAHYAYgBsAEIAaABkAEcAZwBnAFkAQwBJAGsAZQAwAFYAdQBkAGoAcABYAFMARQBsAFUAUgBVAHgASgBVADEAUgBmAFIARQBsAFMAZgBXAEEAaQBJAGcAMABLAEoARwBrAHUAUQAzAEoAbABZAFgAUgBsAFQAbQA5AFgAYQBXADUAawBiADMAYwBnAFAAUwBBAGsAWgBtAEYAcwBjADIAVQBOAEMAaQBSAHAATABrAFoAcABiAEcAVgBPAFkAVwAxAGwASQBEADAAZwBKADMAQgB2AGQAMgBWAHkAYwAyAGgAbABiAEcAdwB1AFoAWABoAGwASgB3ADAASwBKAEcAawB1AFYAWABOAGwAVQAyAGgAbABiAEcAeABGAGUARwBWAGoAZABYAFIAbABJAEQAMABnAEoASABSAHkAZABXAFUATgBDAGkAUgBwAEwAbABaAGwAYwBtAEkAZwBQAFMAQQBuAGMAbgBWAHUAUQBYAE0AbgBEAFEAbwBrAGMAQwBBADkASQBFADUAbABkAHkAMQBQAFkAbQBwAGwAWQAzAFEAZwBVADMAbAB6AGQARwBWAHQATABrAFIAcABZAFcAZAB1AGIAMwBOADAAYQBXAE4AegBMAGwAQgB5AGIAMgBOAGwAYwAzAE0ATgBDAGkAUgB3AEwAbABOADAAWQBYAEoAMABTAFcANQBtAGIAeQBBADkASQBDAFIAcABEAFEAcAAwAGMAbgBrAGcAZQB3ADAASwBJAEMAQQBrAGMAQwA1AFQAZABHAEYAeQBkAEMAZwBwAEQAUQBwADkASQBHAE4AaABkAEcATgBvAEkASABzAE4AQwBpAEEAZwBKAEcAVQBnAFAAUwBBAGsAUgBYAEoAeQBiADMASgBiAE0ARgAwAE4AQwBpAEEAZwBhAFcAWQBnAEsAQwBSAGwASQBDADEAdQBaAFMAQQBrAGIAbgBWAHMAYgBDAEEAdABZAFcANQBrAEkAQwBSAGwATABrAFYANABZADIAVgB3AGQARwBsAHYAYgBpAEEAdABiAG0AVQBnAEoARwA1ADEAYgBHAHcAZwBMAFcARgB1AFoAQwBBAGsAWgBTADUARgBlAEcATgBsAGMASABSAHAAYgAyADQAdQBTAFcANQB1AFoAWABKAEYAZQBHAE4AbABjAEgAUgBwAGIAMgA0AGcATABXADUAbABJAEMAUgB1AGQAVwB4AHMASQBDADEAaABiAG0AUQBnAEoARwBVAHUAUgBYAGgAagBaAFgAQgAwAGEAVwA5AHUATABrAGwAdQBiAG0AVgB5AFIAWABoAGoAWgBYAEIAMABhAFcAOQB1AEwAawA1AGgAZABHAGwAMgBaAFUAVgB5AGMAbQA5AHkAUQAyADkAawBaAFMAQQB0AGIAbQBVAGcASgBHADUAMQBiAEcAdwBwAEkASABzAE4AQwBpAEEAZwBJAEMAQgBsAGUARwBsADAASQBDAFIAbABMAGsAVgA0AFkAMgBWAHcAZABHAGwAdgBiAGkANQBKAGIAbQA1AGwAYwBrAFYANABZADIAVgB3AGQARwBsAHYAYgBpADUATwBZAFgAUgBwAGQAbQBWAEYAYwBuAEoAdgBjAGsATgB2AFoARwBVAE4AQwBpAEEAZwBmAFMAQgBsAGIASABOAGwASQBIAHMATgBDAGkAQQBnAEkAQwBCAGwAZQBHAGwAMABJAEQARQBOAEMAaQBBAGcAZgBRADAASwBmAFEAMABLACIAfQAnACAAfAAgAEMAbwBuAHYAZQByAHQARgByAG8AbQAtAEoAcwBvAG4AKQAuAFMAYwByAGkAcAB0ACkAKQAgAHwAIABpAGUAeAA= -ExecutionPolicy Bypass',
+      {
+        env: {
+          WHITELIST_DIR: nonDefaultFilePath
+            ? `${nonDefaultFilePath}\\Salad\\plugin-bin`
+            : `${process.env.APPDATA}\\Salad\\plugin-bin`,
```

Comment:
If it isn't already, import the `path` module and use the `join` function.

```suggestion
            ? path.join(nonDefaultFilePath, 'Salad/plugin-bin')
            : path.join(process.env.APPDATA, 'Salad/plugin-bin'),
```

Reply:
`process.env.APPDATA` is possibly undefined so we aren't able to use the `path.join`. i can have this default to something else if proces.env.APPDATA is undefined but what should i default it to?

Reply:
It's technically possible to have an undefined APPDATA environment variable, although incredibly unlikely. You could handle it... if it's not defined, you can push an error to the web side.